Google.Api.CommonProtos.Tests/Type/DecimalTest.cs

```diff
@@ -0,0 +1,103 @@
+﻿/*
+ * Copyright 2021 Google LLC All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+using Xunit;
+
+namespace Google.Api.CommonProtos.Tests.Type
+{
+    public class DecimalTest
+    {
+        // Note: we can't use InlineData in the tests with decimal values, as decimals can't be used in attributes.
+        // See https://codeblog.jonskeet.uk/2014/08/22/when-is-a-constant-not-a-constant-when-its-a-decimal/
+        // for details.
+
+        public static TheoryData<string, decimal> RoundtripData = new TheoryData<string, decimal>
+        {
+            { "1.23", 1.23m },
+            { "-1.23", -1.23m },
+            { "5", 5m },
+            { "1.00", 1.00m },
+            { "1.2345678901234567890123456789", 1.2345678901234567890123456789m },
+            { "0.0000000000000000000000000001", 0.0000000000000000000000000001m },
+            { "79228162514264337593543950335", 79228162514264337593543950335m },
+            { "-79228162514264337593543950335", -79228162514264337593543950335m }
+        };
+
+        public static TheoryData<string, decimal> ParseOnlyData = new TheoryData<string, decimal>
+        {
+            // Leading insignificant zeroes
+            { "000.23", 0.23m },
+            // Missing leading zero
+            { ".23", 0.23m },
+            // Exponents
+            { "2.3e4", 23000m },
+            { "2.3e+4", 23000m },
+            { "2.3e-4", 0.00023m },
+            { "2.3E4", 23000m },
+            { "2.3E-4", 0.00023m },
+            { "-2.3e4", -23000m },
+            { "-2.3e-4", -0.00023m },
+            // Explicit positive sign
+            { "+1.23", 1.23m },
```

Comment:
"-0.00"?

Reply:
Ooh, good one. That doesn't round-trip (it ends up as "0.00"). Will add two tests...